### ISSUE TITLE
Error when introducing Linaria via wyw-in-js to Vike's react template

### ISSUE THREAD

**Issue by BillStark001** [vikejs/vike#2039]
### Description

In case some of you are not familiar with Linaria, it is a zero-runtime css-in-js library which I'm in favor of. However, introducing Linaria into a Vike project faces the following issue.

The issue could be reproduced stably by:
1. `npm create vike -- --react`
2. add wyw-in-js and linaria following [this link](https://wyw-in-js.dev/bundlers/vite)
3. create a styled component in any page to be rendered
4. launch the dev server(`npm run dev`, essentially `node ./server/index.ts`), open that page in the browser, refresh it once or twice
5. the issue occurs and never disappears unless the dev server is restarted

Besides, the issue does not occur if a Vite dev server is launched (`vite`).

I have analyzed the issue myself, and here are my findings:

The principle of Linaria is to transform certain codes into CSS classnames in strings while injecting an import statement of a generated css file. The generated file's [name](https://github.com/Anber/wyw-in-js/blob/05d3d7234728a8574792d185a8fcb412696707fa/packages/vite/src/index.ts#L188) ends with `.css`, but later it gets transferred to JS HMR code that patches the stylesheet like 
```
import {
    createHotContext as __vite__createHotContext
} from "/@vite/client"; 
import.meta.hot = __vite__createHotContext("/pages/index/+Page.wyw-in-js.css?direct"); import {
    updateStyle as __vite__updateStyle, removeStyle as __vite__removeStyle
} from "/@vite/client" 
const __vite__id = "/Users/billstark001/Desktop/programs/vike-react-linaria-test/pages/index/+Page.wyw-in-js.css" 
const __vite__css = ".cxoeud8{color:blue;}\n" 
__vite__updateStyle(__vite__id, __vite__css) 
import.meta.hot.accept() 
import.meta.hot.prune(() => __vite__removeStyle(__vite__id))
```
. This confuses Vike's `renderPage` function by emitting an error in a [type assertion](https://github.com/vikejs/vike/blob/40cb3aa4cff0555ca628f2b8555d299759bb606f/vike/node/runtime/renderPage/getPageAssets/retrieveAssetsDev.ts#L67). The generated `.css` files' `ModuleNode`'s' client type is `css` while SSR type is `js`.

The stack trace is as follow:
```
[vike@0.4.211][Bug] You stumbled upon a Vike bug. Go to https://github.com/vikejs/vike/issues/new and copy-paste this error. A maintainer will fix the bug (usually within 24 hours).
    at isStyle (file:///Users/billstark001/Desktop/programs/vike-react-linaria-test/node_modules/vike/dist/esm/node/runtime/renderPage/getPageAssets/retrieveAssetsDev.js:68:9)
    at collectCss (file:///Users/billstark001/Desktop/programs/vike-react-linaria-test/node_modules/vike/dist/esm/node/runtime/renderPage/getPageAssets/retrieveAssetsDev.js:36:9)
    at file:///Users/billstark001/Desktop/programs/vike-react-linaria-test/node_modules/vike/dist/esm/node/runtime/renderPage/getPageAssets/retrieveAssetsDev.js:58:9
    at Set.forEach (<anonymous>)
    at collectCss (file:///Users/billstark001/Desktop/programs/vike-react-linaria-test/node_modules/vike/dist/esm/node/runtime/renderPage/getPageAssets/retrieveAssetsDev.js:57:25)
    at file:///Users/billstark001/Desktop/programs/vike-react-linaria-test/node_modules/vike/dist/esm/node/runtime/renderPage/getPageAssets/retrieveAssetsDev.js:58:9
    at Set.forEach (<anonymous>)
    at collectCss (file:///Users/billstark001/Desktop/programs/vike-react-linaria-test/node_modules/vike/dist/esm/node/runtime/renderPage/getPageAssets/retrieveAssetsDev.js:57:25)
    at file:///Users/billstark001/Desktop/programs/vike-react-linaria-test/node_modules/vike/dist/esm/node/runtime/renderPage/getPageAssets/retrieveAssetsDev.js:23:9
    at async Promise.all (index 0)
    at retrieveAssetsDev (file:///Users/billstark001/Desktop/programs/vike-react-linaria-test/node_modules/vike/dist/esm/node/runtime/renderPage/getPageAssets/retrieveAssetsDev.js:6:5)
    at getPageAssets (file:///Users/billstark001/Desktop/programs/vike-react-linaria-test/node_modules/vike/dist/esm/node/runtime/renderPage/getPageAssets.js:21:21)
    at Object.__getPageAssets (file:///Users/billstark001/Desktop/programs/vike-react-linaria-test/node_modules/vike/dist/esm/node/runtime/renderPage/loadUserFilesServerSide.js:51:36)
    at injectHtmlTagsToString (file:///Users/billstark001/Desktop/programs/vike-react-linaria-test/node_modules/vike/dist/esm/node/runtime/html/injectAssets.js:9:24)
    at renderDocumentHtml (file:///Users/billstark001/Desktop/programs/vike-react-linaria-test/node_modules/vike/dist/esm/node/runtime/html/renderHtml.js:38:32)
    at executeOnRenderHtmlHook (file:///Users/billstark001/Desktop/programs/vike-react-linaria-test/node_modules/vike/dist/esm/node/runtime/renderPage/executeOnRenderHtmlHook.js:30:24)
    at renderPageAlreadyRouted (file:///Users/billstark001/Desktop/programs/vike-react-linaria-test/node_modules/vike/dist/esm/node/runtime/renderPage/renderPageAlreadyRouted.js:58:30)
    at renderPageNominal (file:///Users/billstark001/Desktop/programs/vike-react-linaria-test/node_modules/vike/dist/esm/node/runtime/renderPage.js:286:36)
    at renderPageAlreadyPrepared (file:///Users/billstark001/Desktop/programs/vike-react-linaria-test/node_modules/vike/dist/esm/node/runtime/renderPage.js:116:45)
    at renderPageAndPrepare (file:///Users/billstark001/Desktop/programs/vike-react-linaria-test/node_modules/vike/dist/esm/node/runtime/renderPage.js:96:12)
    at file:///Users/billstark001/Desktop/programs/vike-react-linaria-test/node_modules/vike/dist/esm/node/plugin/shared/getHttpRequestAsyncStore.js:54:35
    at renderPage (file:///Users/billstark001/Desktop/programs/vike-react-linaria-test/node_modules/vike/dist/esm/node/runtime/renderPage.js:43:35)
    at file:///Users/billstark001/Desktop/programs/vike-react-linaria-test/server/index.ts:57:29
```

Though my current work is a static site and thus it doesn't affect me, this issue is reported by the [bug] instruction. Yet, I didn't know if Linaria or Vike should be responsible for this, since the HMR code's type is `css` seems buggy itself.

Last but not least, I have to express my gratitude to the Vike team for your incredible work, especially after being tortured by the whole Next.js ecosystem :D

**Comment by BillStark001** [comment]
A temporary workaround will be applying the following patch:

`vike+0.4.211.patch`
```
diff --git a/node_modules/vike/dist/esm/node/runtime/renderPage/getPageAssets/retrieveAssetsDev.js b/node_modules/vike/dist/esm/node/runtime/renderPage/getPageAssets/retrieveAssetsDev.js
index 0639ba9..7c80835 100644
--- a/node_modules/vike/dist/esm/node/runtime/renderPage/getPageAssets/retrieveAssetsDev.js
+++ b/node_modules/vike/dist/esm/node/runtime/renderPage/getPageAssets/retrieveAssetsDev.js
@@ -60,8 +60,13 @@ function collectCss(mod, styleUrls, visitedModules, importer) {
 }
 function isStyle(mod) {
     if (styleFileRE.test(mod.url) || (mod.id && /\?vue&type=style/.test(mod.id))) {
-        // `mod.type` seems broken
-        assert(mod.type === 'js');
+        if (mod.id.includes('wyw-in-js') && mod.type === 'css') {
+            // special case for Linaria & WYW-in-JS
+            // do nothing
+        } else {
+            // `mod.type` seems broken
+            assert(mod.type === 'js');
+        }
         // logModule(mod)
         return true;
     }
```
by `patch-package`.

**Comment by brillout** [comment]
How about this?

```js
function isStyle(mod) {
  if (mod.type === 'css') return true
  if (styleFileRE.test(mod.url) || (mod.id && /\?vue&type=style/.test(mod.id))) return true
  return false
}
```

**Comment by BillStark001** [comment]
This works for me. 

Actually I don't know the purpose of that assertion (perhaps you want to find cases that violate it?) but as long as you think removing it is fine, there shouldn't be a problem.

**Comment by brillout** [comment]
Pre-released as `0.4.211-commit-be2fe23`.

Yea, exactly, I wanted to know when it isn't `'js'` which is surprisingly only very seldomly the case.
